### PR TITLE
Csv disk index

### PR DESCRIPTION
**Pull request by artemp**

### CODE REVIEW

File: plugins/input/csv/csv_datasource.cpp [hunk 1]
```diff
@@ -596,6 +608,12 @@ mapnik::featureset_ptr csv_datasource::features(mapnik::query const& q) const
                 return std::make_shared<csv_inline_featureset>(inline_string_, locator_, separator_, headers_, ctx_, std::move(index_array));
             }
         }
+        else if (has_disk_index_)
+        {
+            std::cerr << "DISK_INDEX" << std::endl;
```
Comment (springmeyer) [inline plugins/input/csv/csv_datasource.cpp]: @artemp - remove this debug print?

File: plugins/input/csv/csv_index_featureset.cpp [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+mapnik::feature_ptr csv_index_featureset::next()
+{
+    /*
+    if (row_limit_ && count_ >= row_limit_)
+    {
+        return feature_ptr();
+    }
+    */
```
Comment (springmeyer) [inline plugins/input/csv/csv_index_featureset.cpp]: @artemp why is `row_limit` commented?

**Comment by springmeyer** [comment]
@artemp - looking really good / promising. What do you think about a single `mapnik-index` command that works to create `.index` for both shapefiles and csv (and more, like geojson, in the future)? What would it take to provide that?

File: plugins/input/csv/csv_datasource.cpp (hunk 1 above)
Comment (artemp) [inline plugins/input/csv/csv_datasource.cpp]: @springmeyer - yes.

File: plugins/input/csv/csv_index_featureset.cpp (hunk 2 above)
Comment (artemp) [inline plugins/input/csv/csv_index_featureset.cpp]: @springmeyer - do we really need row_limit for disk-index access ? [this-is-work-in-progress]

**Pull request merged**